test/abstract_syntax_tree.coffee

```diff
@@ -10,207 +10,595 @@ getExpressions = (code) ->
+    type: 'Class'
+    variable: value: 'Klass'
+    body:
+      type: 'Block'
+      expressions: []
+
+  testExpression 'class child extends parent',
+    type: 'Class'
+    variable: value: 'child'
+    parent: value: 'parent'
+    body:
+      type: 'Block'
+      expressions: []
+
+  testExpression 'class Klass then constructor: ->',
+    type: 'Class'
+    variable: value: 'Klass'
+    parent: undefined
+    body:
+      type: 'Value'
+      properties: []
+      base:
+        type: 'Obj'
+        generated: yes
+        properties: [
+          variable: value: 'constructor'
+          value: type: 'Code'
+        ]
 
 test "AST as expected for ExecutableClassBody node", ->
-  # TODO
+  code = """
+    class Klass
+      privateStatic = if 42 then yes else no
+      getPrivateStatic: -> privateStatic
+    """
+  testExpression code,
+    type: 'Class'
+    variable: value: 'Klass'
+    body:
+      type: 'Block'
+      expressions: [
+        type: 'Assign'
+        variable: value: 'privateStatic'
+        value: type: 'If'
+      ,
+        type: 'Obj'
+        generated: true
+        properties: [
+          type: 'Assign'
+          variable: value: 'getPrivateStatic'
+          value:
+            type: 'Code'
+            body:
+              type: 'Value'
+              properties: []
+        ]
+      ]
 
 test "AST as expected for ModuleDeclaration node", ->
-  # TODO
+  testExpression 'export {X}',
+    clause: specifiers: [
+      type: 'ExportSpecifier'
+      moduleDeclarationType: 'export'
+      identifier: 'X'
+    ]
+
+  testExpression 'import X from "."',
+    clause: defaultBinding:
+      type: 'ImportDefaultSpecifier'
+      moduleDeclarationType: 'import'
+      identifier: 'X'
 
 test "AST as expected for ImportDeclaration node", ->
-  # TODO
+  testExpression 'import React, {Component} from "react"',
+    type: 'ImportDeclaration'
+    source:
+      type: 'StringLiteral'
+      value: '"react"'
+      originalValue: 'react'
 
 test "AST as expected for ImportClause node", ->
-  # TODO
+  testExpression 'import React, {Component} from "react"',
+    clause:
+      type: 'ImportClause'
+      defaultBinding: type: 'ImportDefaultSpecifier'
+      namedImports: type: 'ImportSpecifierList'
 
 test "AST as expected for ExportDeclaration node", ->
-  # TODO
+  testExpression 'export {X}',
+    clause: specifiers: [
+      type: 'ExportSpecifier'
+      moduleDeclarationType: 'export'
+      identifier: 'X'
+    ]
 
 test "AST as expected for ExportNamedDeclaration node", ->
-  # TODO
+  testExpression 'export fn = ->',
+    type: 'ExportNamedDeclaration'
+    clause:
+      type: 'Assign'
+      variable: value: 'fn'
+      value: type: 'Code'
 
 test "AST as expected for ExportDefaultDeclaration node", ->
-  # TODO
+  testExpression 'export default class',
+    type: 'ExportDefaultDeclaration'
+    clause: type: 'Class'
 
 test "AST as expected for ExportAllDeclaration node", ->
   # TODO
```

Comment:
We could add this one, couldn’t we?

Reply:
Unfortunately I have no idea what it could be.

Reply:
```coffee
export * from "module-name"
```